src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -72,11 +85,12 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 	) {
 		super();
 
-		this._register(this._xterm.onData(e => this._handleInput(e)));
 		this._register(Event.any(
-			this._xterm.onWriteParsed,
 			this._xterm.onCursorMove,
+			this._xterm.onData,
```

Comment:
Since onData has multiple handlers, is there a consistent order in which the events are handled, or does it not matter in this case?

Reply:
It doesn't matter since `_sync` is `throttle(0)`